src/test/ui/macros/issue98466.stderr

```diff
@@ -0,0 +1,14 @@
+error: named argument is not used by name
+  --> $DIR/issue98466.rs:4:26
+   |
+LL |     println!("_x is {}", _x = 5);
+   |                          ^^
```

Comment:
Ideally the end result would look something like this:

```suggestion
error: named string formatting argument is not used by name
  --> $DIR/issue98466.rs:4:26
   |
LL |     println!("_x is {}", _x = 5);
   |                     --   ^^ this named argument is only referred to by position in the formatting string
   |                     |
   |                     this formatting argument uses named argument `_x` by position
   |
   = note: lint `#[deny(named_formatting_arg_by_pos)]` enabled by default
help: use the named argument by name to avoid ambiguity
   |
LL |     println!("_x is {_x}", _x = 5);
   |                      ++
```

Reply:
That seems reasonable to me! Still working on getting the lint to actually be emitted.

Reply:
What I ended up with looks a little different -- since I used `span_suggestion` the help part got rolled in with the suggestion. Should I be using a different method?

Also, the lint is currently `Warn` instead of `Deny`, since it seems like it's been valid to use named arguments positionally for some years. Would `Deny` be better?